### PR TITLE
Add a fast training tutorial

### PR DESCRIPTION
**Pull request by stephenroller**
**Patch description**
Add a tutorial containing tricks on how to train models faster.

**Testing steps**
Read it.

### CODE REVIEW

**Comment by dexterju** [comment]
Did you revoke my review after the email was sent? lol

**Comment by stephenroller** [comment]
Yes, I decided Diana and Jing would be better reviewers.

File: docs/source/tutorial_fast.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+speedup with larger models, with models of >300M often getting a ~50% speedup.
+With the increased batch size, this can often be brought to 2.5x faster.
+
+## Use multiple GPUs
+
+If you have multiple GPUs, you can utilize them by switching from `train` to
+`multiprocessing_train`. If you have 4 GPUs, you'll find training should be
+roughly 3.5x faster. The arguments for the training are left otherwise the same.
+
+    parlai multiprocessing_train \
```
Comment (jxmsML) [inline docs/source/tutorial_fast.md:132]: This does not work as well due to the naming of model multiprocess_train,  calling this 
 https://github.com/facebookresearch/ParlAI/blob/master/parlai/utils/strings.py#L67 will produce the error message:  ``` module 'parlai.scripts.multiprocessing_train' has no attribute 'MultiprocessingTrain' ``` whereas the actual class name is ``` MultiProcessingTrain ```. I would suggest we rename multiprocessing_train as multi_processing_train

Comment (jxmsML) [reply docs/source/tutorial_fast.md:132]: t

Comment (stephenroller) [reply docs/source/tutorial_fast.md:132]: It works after #2772 lands.

File: docs/source/tutorial_fast.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+Add `--dynamic-batching full` (`-dynb full`) to your training command. Note
+that in order to use dynamic batching, we must also set a `--truncate` option.
+We'll use 256, since that is longer than almost all the conversations in our
+data.
+
+    parlai train -df dictfile -m transformer/generator -t convai2 -eps 1.0 -bs 64 \
+        --embedding-size 250 --ffn-size 1000 --num-layers 8 -opt adam -lr 1e-3 \
+        --skip-generation true \
+        --eval-batchsize 128 \
+        --dynamic-batching true --truncate 256
```
[the remark this answers is not in the document]
Comment (jxmsML) [reply docs/source/tutorial_fast.md:75]: LGTM except nit: should change the command to ```--dynamic-batching full``` to be consistent with your instructions above? along with the other commands below. There is also a typo ``` --dynamic-batching trufull``` in L137.

File: docs/source/tutorial_fast.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+usually increase your learning rate pretty substantially when using dynamic
+batching, to compensate for the fewer steps.
+
+## FP16
+
+If you have access to an NVIDIA GPU with FP16 CUDA Cores (V100, GTX 2080, etc),
+then you can get large speedups by switching on the option `--fp16 true`. The
+default version of FP16 requires that you install
+[APEX](https://github.com/NVIDIA/apex), but you can use a simplified version
+(which doesn't depend on APEX) with `--fp16 true --fp16-impl mem_efficienet`.
```
Comment (emilydinan) [inline docs/source/tutorial_fast.md:96]: spelling: mem_efficient

**Pull request merged**